### PR TITLE
run checkov on plan file

### PR DESCRIPTION
**Pull request by kwong0222**
For accuracy, run static analysis on the plan output instead of source code. If analysis is only done on source code, we may get false 
positives. For example, the `aws_backup_vault` resource by default attaches an AWS KMS key to encrypt the data. 

Running the check against source code only results in the check 'Ensure Backup Vault is encrypted at rest using KMS' to be flagged since the ARN 
of the KMS key is not explicitly defined.

Unfortunately, due to limitations on Github ability to reuse steps, generating the plan for checkov results in a lot of duplication. I decided 
against adding 
Checkov to `plan.yml` because this may result in terraform repos using this workflow to fail, which may block merging.

- Add checkov static analysis (#15)
- Create plan file in checkov action
- Add workspace in comment

### CODE REVIEW

File: .github/workflows/checkov.yml [hunk 1]
```diff
@@ -26,13 +48,70 @@ jobs:
...
+          # TODO: Allow for optional override
+          backend_file=backend/${{ inputs.profile }}.tfvars
+          echo "::set-output name=backend_file::$backend_file"
+          terraform init -backend-config "$backend_file"
+
+      - name: Terraform Validate
+        id: validate
+        run: terraform validate -no-color
+
+      - name: Terraform Plan
```
Comment (davegallant) [inline .github/workflows/checkov.yml:91]: Could this potentially collide with the other terraform plan that goes on?

Would it make sense to add it to the existing workflow with an optional toggle?

Comment (kwong0222) [reply .github/workflows/checkov.yml:91]: @davegallant, 

I think I ran into that problem yesterday actually. I added `concurrency: terraform` to the checkov workflow, but not sure if that would resolve the problem. 

The optional toggle makes sense too, that way we won't need to run a second plan. Now that I think about it, maybe there are edge cases where the two plans differ.

How were you thinking of adding the optional toggle? I did a quick search and there doesn't look like a feature in GHA that would provide it out of the box. 

I was thinking something like having a bool input and then in the checkov stage do a `if:`
```
inputs:
... run_checkov:
....

checkov:
 if: inputs.run_checkov == 'true'
```

Comment (davegallant) [reply .github/workflows/checkov.yml:91]: An input bool might be the only way to add a toggle (GHA isn't super flexible).

Don't think it's possible to call a reusable workflow from within a reusable workflow, so workflows could potentially become monolithic.

Comment (kwong0222) [reply .github/workflows/checkov.yml]: Added checkov as part of TF plan.

**Review by davegallant — APPROVED** [review]

**Pull request merged**